x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -99,42 +96,72 @@ interface ExecuteActionHostResponseOutputProps {
 }
 
 export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
-  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize }) => {
+  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize = 'xs' }) => {
+    const { canWriteExecuteOperations } = useUserPrivileges().endpointPrivileges;
+
+    const [executeOutputContent, setExecuteOutputContent] = useState<
+      undefined | ResponseActionExecuteOutputContent
+    >(undefined);
+
     const outputContent = useMemo(
       () =>
         action.outputs &&
         action.outputs[agentId] &&
         (action.outputs[agentId].content as ResponseActionExecuteOutputContent),
-      [agentId, action]
+      [action.outputs, agentId]
     );
 
+    const { data: actionDetails, isFetching } = useGetActionDetails(action.id, {
+      enabled: canWriteExecuteOperations && !outputContent,
+    });
+
+    useEffect(() => {
+      if (
+        !isFetching &&
+        actionDetails &&
+        actionDetails.data &&
+        actionDetails.data.outputs &&
+        actionDetails.data.outputs[agentId]
+      ) {
+        setExecuteOutputContent(
+          actionDetails.data.outputs[agentId].content as ResponseActionExecuteOutputContent
+        );
+      }
+
+      return () => {
+        setExecuteOutputContent(undefined);
+      };
+    }, [actionDetails, agentId, isFetching]);
+
+    if (!canWriteExecuteOperations || !executeOutputContent) {
+      return null;
+    }
+
+    if (isFetching) {
+      return (
+        <EuiSkeletonText size="relative" lines={2} data-test-subj={`${dataTestSubj}-loading`} />
+      );
+    }
+
     return (
-      <EuiFlexGroup direction="column" data-test-subj={dataTestSubj}>
-        <EuiFlexItem>
-          <ResponseActionFileDownloadLink
-            action={action}
-            buttonTitle={EXECUTE_FILE_LINK_TITLE}
+      executeOutputContent && (
```

Comment:
If you are going to keep the API call in this component, then you are missing logic to handle/display API errors

Reply:
Indeed. I am not handling API errors yet. I'll add that.

Reply:
416154822fd34efd8e64bab195c72ff70fc9926e